bokeh/io.py

```diff
@@ -242,17 +248,21 @@ def _show_with_state(obj, state, browser, new):
 
     if state.notebook:
         _show_notebook_with_state(obj, state)
-
     elif state.session_id and state.server_url:
         _show_server_with_state(obj, state, new, controller)
-
-    if state.file:
+    elif state.file:
```

Comment:
I think the code was `if` rather than `elif` on purpose before, that is, output to both server and file at the same time is supposed to work. (I'm not advocating that, but it's my impression of the previous code's intent, and to unwind the intent there are more changes to be made I think - the docs are wrong for example.)